### DISCUSSION TITLE
1.3 定义新的函数

### DISCUSSION THREAD

**Discussion by giscus** [csfive/composing-programs-zh#93] [Announcements]
# 1.3 定义新的函数

SICP Python 中文翻译

https://composingprograms.netlify.app/1/3

**Comment by wsfight** [comment]
感谢译者的付出！

**Comment by ruihao-yan** [comment]
frame应该翻译为框架还是帧呢

**Reply by 1224267419** [reply]
栈帧,这个说法用的多

[a comment by an automated account is not shown]

**Comment by MotingLiuu** [comment]
"调用用户定义的函数会引入局部帧（local frame），它只能访问该函数。"
这个翻译有问题吧，应该是“只能被这个函数访问”

**Comment by huahai27** [comment]
打卡
个人理解 **名称求解（Name Evaluation）** 中：**最早找到的含有该名称的帧** ，是否为 **距离最近的可找到的该名称的桢** ，初次理解时 **最早** 以为是从global frame开始一帧一帧往下开始寻找，欢迎指正

**Reply by kingsunone1** [reply]
我觉得应该是距离最近的可找到的该名称的帧，并且这个最近应该是往上找的（即往全局帧的方向）

**Reply by howwilli** [reply]
并不是从global 开始往下找，具体可以看1.3.3的例子。x有多个不同的值，当调用square(12)时，如果从global开始查找,x等于5,与本义不符。要真正来理解这句话可询问ai，或者学到后面再回顾就理解了

**Comment by BleedingRivers** [comment]
谢谢译者！！！

**Comment by BleedingRivers** [comment]
太抽象了，不过能懂就好

**Comment by Yoisaki511** [comment]
1.3.3节中“Python 通过引入另一个将将 x 与 5 绑定的局部帧来调用 square 函数。“这一句的表述貌似有点奇怪？不知能否删去重复的”将“字？

**Comment by BleedingRivers** [comment]
您是否发错人了？

---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2025年4月27日(周日) 中午12:40
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [csfive/composing-programs-zh] 1.3 定义新的函数 (Discussion #93)




 
1.3.3节中“Python 通过引入另一个将将 x 与 5 绑定的局部帧来调用 square 函数。“这一句的表述貌似有点奇怪？不知能否删去重复的”将“字？

**Comment by xbsheng** [comment]
done

**Comment by czunique** [comment]
打卡，同时感谢译者的付出！网页做的太好了

**Comment by PscreenSarctic** [comment]
打卡喵，谢谢大佬们

**Comment by Smoxiao** [comment]
打卡

**Comment by wanziyawanzi** [comment]
1.3.2 名称求值（Name Evaluation）：名称会求值为当前环境中“最早”包含该名称的帧内所绑定的值。实际含义是否应为：名称求值会使用当前环境链中最内层包含该名称的帧里的绑定值。强调空间的内外，“最早”在中文语境中暗示了时间的先后。
